unity-ggjj/Assets/Scripts/UI/PenaltyManager.cs

```diff
@@ -1,27 +1,68 @@
 using System.Collections.Generic;
 using UnityEngine;
+using UnityEngine.Events;
 
-public class PenaltyManager : MonoBehaviour
+public class PenaltyManager : MonoBehaviour, IPenaltyManager
 {
+    [Tooltip("Drag a DirectorActionDecoder here")]
+    [SerializeField] private DirectorActionDecoder _directorActionDecoder;
+
+    [Tooltip("Drag a NarrativeScriptPlaylist here")]
+    [SerializeField] private NarrativeScriptPlaylist _narrativeScriptPlaylist;
+    
     [Tooltip("Drag the prefab for the penalty UI object here.")]
     [SerializeField]private Animator _penaltyObject;
 
+    [SerializeField] private int _penaltyCount = 5;
+
+    [Header("Events")]
+    [SerializeField] private UnityEvent<NarrativeScript> _onPenaltiesDepleted;
+    
     private readonly Queue<Animator> _penaltyObjects = new Queue<Animator>();
 
     public int PenaltiesLeft => _penaltyObjects.Count;
 
+    private void Awake()
+    {
+        _directorActionDecoder.Decoder.PenaltyManager = this;
+    }
+    
     /// <summary>
-    /// Creates a specified number of penalties.
+    /// Creates penalty UI objects on examination start
     /// </summary>
-    /// <param name="penaltyCount">The number of penalties to create.</param>
-    public void Initialize(int penaltyCount)
+    public void OnCrossExaminationStart()
     {
-        for (int i = 0; i < penaltyCount; i++)
+        gameObject.SetActive(true);
+        if (_penaltyObjects.Count == 0)
         {
-            _penaltyObjects.Enqueue(Instantiate(_penaltyObject, transform));
+            for (int i = 0; i < _penaltyCount; i++)
+            {
+                _penaltyObjects.Enqueue(Instantiate(_penaltyObject, transform));
+            }
         }
     }
-    
+
+    /// <summary>
+    /// Hide the penalty UI objects on cross examination end
+    /// </summary>
+    public void OnCrossExaminationEnd()
+    {
+        gameObject.SetActive(false);
+    }
+
+    /// <summary>
+    /// Destroy all penalty objects
+    /// </summary>
+    public void ResetPenalties()
+    {
+        while (_penaltyObjects.Count > 0)
+        {
+            var penalty = _penaltyObjects.Dequeue();
+            Destroy(penalty.gameObject);
+        }
```

Comment:
Nit: I'm totally fine if this gets dismissed as personal preference, but `while`-loops with implied exit conditions and no timeout/fallback exit condition scare me. 👀  
Implied in the sense, that `.Count` is never **directly** changed in the body (i.e. `_penaltyObjects.Count--`, which is impossible given the classes public interface) but only as a side-effect of `.Dequeue()`.  
And "no fallback" along the lines of "ran more than 10000 times for whatever reason? break! to guarantee the process never freezes".

I personally would probably prefer safely looping over the enumerable first and then clearing it afterwards instead:
```suggestion
        foreach (var penaltyAnimator in _penaltyObjects)
        {
            Destroy(penaltyAnimator.gameObject);
        }
        _penaltyObjects.Clear();
```